src/Rips_complex/include/gudhi/Sparse_rips_complex.h

```diff
@@ -131,30 +187,32 @@ class Sparse_rips_complex {
   template <typename Distance>
   void compute_sparse_graph(Distance& dist, double epsilon, Filtration_value mini, Filtration_value maxi) {
     const auto& points = sorted_points; // convenience alias
-    const int n = boost::size(points);
+    Vertex_handle n = boost::size(points);
     double cst = epsilon * (1 - epsilon) / 2;
-    graph_.~Graph();
-    new (&graph_) Graph(n);
-    // for(auto v : vertices(g)) // doesn't work :-(
-    typename boost::graph_traits<Graph>::vertex_iterator v_i, v_e;
-    for (std::tie(v_i, v_e) = vertices(graph_); v_i != v_e; ++v_i) {
-      auto v = *v_i;
-      // This whole loop might not be necessary, leave it until someone investigates if it is safe to remove.
-      put(vertex_filtration_t(), graph_, v, 0);
+    max_v = -1; // Useful for the size of the map lambda.
+    for (Vertex_handle i = 0; i < n; ++i) {
+      if ((params[i] < mini || params[i] <= 0) && i != 0) break;
+      // The parameter of the first point is not very meaningful, it is supposed to be infinite,
+      // but if the type does not support it...
+      // It would be better to do this reduction of the number of points earlier, around choose_n_farthest_points.
+      add_vertex(points[i], graph_);
+      max_v = std::max(max_v, points[i]);
     }
+    n = num_vertices(graph_);
 
     // TODO(MG):
     // - make it parallel
     // - only test near-enough neighbors
-    for (int i = 0; i < n; ++i) {
+    for (Vertex_handle i = 0; i < n; ++i) {
       auto&& pi = points[i];
       auto li = params[i];
-      if (li < mini) break;
-      for (int j = i + 1; j < n; ++j) {
+      // If we inserted all the points, points with multiplicity would get connected to their first representative,
+      // no need to handle the redundant ones in the outer loop.
+      // if (li <= 0 && i != 0) break;
+      for (Vertex_handle j = i + 1; j < n; ++j) {
         auto&& pj = points[j];
         auto d = dist(pi, pj);
```

Comment:
In src/Rips_complex/example/sparse_rips_complex.cpp, we have:
```cpp
  using Point = std::vector<double>;
  using Simplex_tree = Gudhi::Simplex_tree<Gudhi::Simplex_tree_options_fast_persistence>;
  using Filtration_value = Simplex_tree::Filtration_value;
```
As points are a vector of doubles, the distance function will return a `double` value, whereas `Gudhi::Simplex_tree_options_fast_persistence::Filtration_value` is a `float`.
The fix should be fixing the example.

Reply:
Can you explain what the problem is? We have a distance function that gives a precise result, we have a filtration where we are happy with a lower precision, so we drop some precision in the middle, I think that works...

Reply:
It is just a matter of warnings fix, but here, we use points as vector of double. The distance from 2 points will be saved in a double, whereas the filtration values in the simplex tree will be saved as a float.
I think we should use vector of float for a point, or use a double for simplex tree filtration value type.

Reply:
That clearly predates this patch, so we can handle it separately.
I believe it is a perfectly valid use case. I understand why some compilers warn when converting from double to float, but in some cases we may want to do that conversion.
We could add some explicit casts to remove the warning, but it isn't really our place to decide if dropping precision is wanted, it is up to the user. Which tends to favor disabling the warning in the example with a `#pragma`.
We could choose to have an example that does not convert so we don't have to see the warning in the testsuite. That's a cop out, but well...

Reply:
We can keep it like that, I understand your point, no problem.